src/cli/self_update/unix.rs

```diff
@@ -57,66 +57,50 @@ pub fn complete_windows_uninstall() -> Result<utils::ExitCode> {
     panic!("stop doing that")
 }
 
-pub fn do_remove_from_path(methods: &[PathUpdateMethod]) -> Result<()> {
-    for method in methods {
-        if let PathUpdateMethod::RcFile(ref rcpath) = *method {
-            let file = utils::read_file("rcfile", rcpath)?;
-            let addition = format!("\n{}\n", shell_export_string()?);
+pub fn do_remove_from_path() -> Result<()> {
+    for sh in shell::get_available_shells() {
+        let source_cmd = format!("\n{}\n", sh.source_string()?);
+        let source_bytes = source_cmd.into_bytes();
 
+        // Check more files for cleanup than normally are updated.
+        for rc in sh.rcfiles().iter().filter(|rc| rc.is_file()) {
+            let file = utils::read_file("rcfile", &rc)?;
             let file_bytes = file.into_bytes();
-            let addition_bytes = addition.into_bytes();
-
-            let idx = file_bytes
-                .windows(addition_bytes.len())
-                .position(|w| w == &*addition_bytes);
-            if let Some(i) = idx {
-                let mut new_file_bytes = file_bytes[..i].to_vec();
-                new_file_bytes.extend(&file_bytes[i + addition_bytes.len()..]);
-                let new_file = String::from_utf8(new_file_bytes).unwrap();
-                utils::write_file("rcfile", rcpath, &new_file)?;
-            } else {
-                // Weird case. rcfile no longer needs to be modified?
+            if let Some(idx) = file_bytes
```

Comment:
It would perhaps be nice to still work if someone added whitespace by mistake. Thats a separate enhancement obviously, but maybe a FIXME?